app/controllers/hats_controller.rb

```diff
@@ -62,4 +64,46 @@ def reject_request
 
     redirect_to "/hats/requests"
   end
+
+  def doff
+    @title = "Doffing a Hat"
+  end
+
+  def doff_by_user
+    if doff_with_reason(@hat, params[:reason])
+      redirect_to @user
+    else
+      redirect_to doff_hat_path(@hat)
+    end
+  end
+
+  private
+
+  def doff_with_reason(hat, reason)
```

Comment:
Merge this into the `doff_by_user` action. It's not likely to be reused anywhere and it'd be nice to minimize the verbosity of having a separate return value from it.